crates/cxx-qt-lib/src/core/qmargins.rs

```diff
@@ -0,0 +1,102 @@
+// SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+use cxx::{type_id, ExternType};
+use std::fmt;
+
+#[cxx::bridge]
+mod ffi {
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/qmargins.h");
+        type QMargins = super::QMargins;
+        include!("cxx-qt-lib/qstring.h");
+        type QString = crate::QString;
+
+        /// Returns the bottom margin.
+        fn bottom(self: &QMargins) -> i32;
+
+        /// Returns true if all margins are is 0; otherwise returns false.
+        #[rust_name = "is_null"]
+        fn isNull(self: &QMargins) -> bool;
+
+        /// Returns the left margin.
+        fn left(self: &QMargins) -> i32;
+
+        /// Returns the right margin.
+        fn right(self: &QMargins) -> i32;
+
+        /// Sets the bottom margin to bottom.
+        #[rust_name = "set_bottom"]
+        fn setBottom(self: &mut QMargins, bottom: i32);
+
+        /// Sets the left margin to left.
+        #[rust_name = "set_left"]
+        fn setLeft(self: &mut QMargins, left: i32);
+
+        /// Sets the right margin to right.
+        #[rust_name = "set_right"]
+        fn setRight(self: &mut QMargins, right: i32);
+
+        /// Sets the Top margin to Top.
+        #[rust_name = "set_top"]
+        fn setTop(self: &mut QMargins, top: i32);
+
+        /// Returns the top margin.
+        fn top(self: &QMargins) -> i32;
+    }
+
+    #[namespace = "rust::cxxqtlib1"]
+    unsafe extern "C++" {
+        include!("cxx-qt-lib/common.h");
+
+        #[doc(hidden)]
+        #[rust_name = "qmargins_default"]
+        fn construct() -> QMargins;
+        #[doc(hidden)]
+        #[rust_name = "qmargins_new"]
+        fn construct(left: i32, top: i32, right: i32, bottom: i32) -> QMargins;
+        #[doc(hidden)]
+        #[rust_name = "qmargins_to_qstring"]
+        fn toQString(value: &QMargins) -> QString;
+    }
+}
+
+/// The QMargins class defines the four margins of a rectangle.
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[repr(C)]
+pub struct QMargins {
```

Comment:
We also don't have QSet/QVector support for these yet.
We should probably decide on whether to add everything for these containers for every type in cxx-qt-lib or only select ones.

QVariant I think we should support for everything. But i.e. `QSet<QMargins>` isn't really a common use-case...

Reply:
So QMargins(F) is not in QVariant in Qt, so i wondered if we should bother adding it too ? Maybe I can add it anyway...

And QMargins(F) is not hashable so shouldn't be in QSet/QHash, but could be added to QList/QVector.

Reply:
So adding support in QVariant requires that we declare `Q_DECLARE_METATYPE` for `QMargins` which doesn't exist by default, otherwise `canConvert` doesn't work. So should we declare this or skip it like Qt does?

Reply:
If Qt doesn't have it, skip it.
I just didn't know Qt didn't support it by default xD.

Then I think it's also not worth the compile-time tradeoff to support these with QList/QVector if Qt doesn't even bother adding them to QVariant.